File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        maxAvg=0
        for j in range(k):
            maxAvg+=nums[j]
        
        r=k
        sum1=maxAvg
        while r < len(nums):

            
            sum1=sum1-nums[r-k]+nums[r]
            if sum1 > maxAvg:
                maxAvg=sum1
            r+=1
            
        return maxAvg/float(k)


        

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int sum = 0;
        int curSum = 0;
        int l = 0;
        for(int i = 0; i < k; i++){
            sum += nums[i];
        }
        curSum = sum;
        for(int r = k; r < nums.length; r++){
            curSum += nums[r];
            curSum -= nums[l];
            l++;
            sum = Math.max(sum,curSum);

        }
        return sum/(double)k;

        
    }
}